nntrainer/layers/layer.cpp

```diff
@@ -84,6 +100,74 @@ void Layer::copy(std::shared_ptr<Layer> l) {
   this->num_outputs = l->num_outputs;
 }
 
+sharedConstTensors Layer::forwarding_with_val(sharedConstTensors input,
+                                   sharedConstTensors in) {
+
+  for(unsigned int i=0;i<num_inputs;++i){
+    net_input[i]->var = *input[i];
+  }
+
+  if (num_outputs == 0)
+    throw("invalid number of outputs");
+
+  if (num_outputs != net_hidden.size())
```

Comment:
same, this would be verified in initialize()

Reply:
done.

Reply:
I thought net_hidden is the structure managed by the graph.

Reply:
> I thought net_hidden is the structure managed by the graph.

Yes. you are right. but this function is only the unittest which is not use graph structure. Directly call with input.